ctk/common/src/main/kotlin/org/codice/compliance/verification/core/CoreVerifier.kt

```diff
@@ -104,25 +105,37 @@ class CoreVerifier(val node: Node) {
                         node = node)
         }
 
-        fun decryptAndValidateSchema(responseDom: Node) {
+        private fun preProcess(responseDom: Node) {
```

Comment:
✏️ While it's not very much better, I think this recursive version of this function is a little clearer.
```kotlin
        private fun preProcess(responseDom: Node,
                                encVerifier: EncryptionVerifier = EncryptionVerifier()) {
            val encElements = retrieveCurrentEncryptedElements(responseDom)
            if (encElements.isEmpty()) {
                Log.debugWithSupplier {
                    "Decrypted SAML Response:\n\n ${responseDom.prettyPrintXml()}"
                }
                return
            }

            Log.debugWithSupplier {
                "Starting a pass of decryption and schema validation" +
                        " on the SAML Response."
            }

            SchemaValidator.validateSAMLMessage(responseDom)
            encVerifier.verifyAndDecryptResponse(encElements)
            preProcess2(responseDom, encVerifier)
        }
```

Reply:
I like this better too. Fixed.